sdk/testing/src/main/java/io/opentelemetry/sdk/testing/exporter/InMemoryMetricReader.java

```diff
@@ -52,13 +55,83 @@ public class InMemoryMetricReader implements MetricReader {
   private final DefaultAggregationSelector defaultAggregationSelector;
   private final AtomicBoolean isShutdown = new AtomicBoolean(false);
   private volatile MetricProducer metricProducer = MetricProducer.noop();
+  private final MemoryMode memoryMode;
+
+  public static class InMemoryMetricReaderBuilder {
```

Comment:
We've been making builders top level calsses rather than inner classes in this repo. Also, we make the constructor package private, and have a static accessor to initialize a builder in the target class. Usage looks like:
```
InMemoryMetricReader.builder().setAggregationSelector(...).build()
```